products/memgraph/upcoming/tutorials/got-deaths.md

```diff
@@ -0,0 +1,232 @@
+
+```opencypher
+MATCH (c:Character {name: 'Daenerys Targaryen'})-[k:Killed]->(c2:Character)
+MATCH (c)-[:KillerIn]->(d:Death)<-[:VictimIn]-(c2)
+MATCH (d)-[:InEpisode]-(e:Episode)
+return distinct(c2.name) as Victim, COUNT(d) as Kills, e.name
+ORDER BY Kills DESC
+```
+
+7) Remember that shocking last episode of the fifth season when they killed Jon Snow and we totally thought
+he was gonna stay dead? Well, let's list all the characters that would survive if he actually stayed dead.
+
+```opencypher
+MATCH (c:Character {name: 'Jon Snow'})-[k:Killed]->(c2:Character)
+MATCH (c)-[:VictimIn]->(d2:Death)
+MATCH (c)-[:KillerIn]->(d:Death)<-[:VictimIn]-(c2)
+WHERE d.order>d2.order
+return distinct(c2.name) as Victim, COUNT(d) as Kills
+ORDER BY Kills DESC
+```
+
+8) Houses or allegiances are one of the main aspects of Westeros. We know some houses killed more characters
+than others, but that doesn't matter in the end, what matters is efficiency. Let's list the allegiances with
+the best Kill/Death Ratios. Here we came across one more problem, if an allegiance had more deaths than kills,
+the KD would be 0. This can easily be fixed with the `toFloat()` function.
+
+```opencypher
+MATCH (c:Character)-[death:Killed]->(c2:Character)-[l:LoyalTo]->(a:Allegiance)
+WITH a, sum(death.count) as Deaths
+MATCH (c:Character)<-[kill:Killed]-(c2:Character)-[l:LoyalTo]->(a:Allegiance)
+RETURN a.name, sum(kill.count) as Kills, Deaths, ROUND(100 *(toFLoat(sum(kill.count))/Deaths))/100 as KD
+ORDER BY KD DESC;
+```
+
+9) One of the best-rated episodes, Battle of the Bastards, showed us a fight between two houses: Stark and Bolton.
+Let's see which one had more casualties.
+
+```opencypher
+MATCH (c:Character)-[:LoyalTo]->(a:Allegiance)
+MATCH (c)-[:VictimIn]-(d:Death)-[:InEpisode]-(e:Episode {name: 'Battle of the Bastards'})
+RETURN a.name AS HouseName, COUNT(d) AS DeathCount
+ORDER BY DeathCount DESC
+LIMIT 2;
+```
+
+10) Who do you think was the biggest traitor in terms of killing in its own allegiance? Well, let's check it out!
+
+```opencypher
+MATCH (c:Character)-[:Killed]->(c2:Character)
+MATCH (c)-[:LoyalTo]->(a:Allegiance)<-[:LoyalTo]-(c2)
+RETURN c.name AS Traitor, COUNT(c2) AS Kills
+ORDER BY Kills DESC
+```
+
+11) One of the biggest features of Memgraph is drawing the graphs of queries we execute. Let's visualize all the
+Loyalties with Characters. Execute the following query and head out to `GRAPH` tab.
+
+```opencypher
+match (c:Character)-[i:LoyalTo]-(l) return c,i,l;
+```
+
+12) Let's see how it looks like if we want to visualize Jon Snow kills with its locations.
+
+```opencypher
+MATCH (c:Character {name: 'Jon Snow'})-[k:Killed]->(c2:Character)
+MATCH (c)-[:KillerIn]->(d:Death)<-[di:VictimIn]-(c2)
+MATCH (d)-[i:InLocation]->(l:Location)
+return c2,di,d,i,l
+```
+
+13) Memgraph supports graph algorithms as well. Let's use Dijkstra's shortest path algorithm to show the most
+gruesome path of kills.

```

Comment:
Explain what a path of kills is. Otherwise, it is left for a reader to define it.